### PR TITLE
[WebNN EP] Release WebNN MLGraphBuilder after Compile to free memory

### PR DESCRIPTION
**Pull request by Honry**
This would help release the constants bound by the MLGraphBuilder.

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

**Comment by Honry** [comment]
cc/ @huningxin

**Review by guschmue — APPROVED** [review]

File: onnxruntime/core/providers/webnn/webnn_execution_provider.cc [hunk 1]
```diff
@@ -337,6 +344,9 @@ common::Status WebNNExecutionProvider::Compile(const std::vector<FusedNodeAndGra
     node_compute_funcs.push_back(compute_info);
   }
 
+  // Explictly release the WebNN builder to free memory.
```
Comment (fdwr) [inline onnxruntime/core/providers/webnn/webnn_execution_provider.cc:347]:
```suggestion
  // Explicitly release the WebNN builder to free memory.
```
Whenever you're in the area next time...

**Review by fdwr — COMMENTED** [review]
👍

**Pull request merged**